test/extended/operators/olm.go

```diff
@@ -168,6 +168,28 @@ var _ = g.Describe("[sig-operator] an end user can use OLM", func() {
 	g.It("can subscribe to the operator", func() {
 		g.By("Cluster-admin user subscribe the operator resource")
 
+		// skip test if redhat-operators is not present or disabled
+		sources, err := oc.AsAdmin().Run("get").Args("operatorhub/cluster", "-o", "jsonpath='{.spec.sources}'").Output()
```

Comment:
The logic here is a little bit more complex. In addition to checking if redhat-operators is explicitly disabled in `spec.Sources`, you also need to check if `spec.disableAllDefaultSources` is true, and if it is check `if source.Name == "redhat-operators" && source.Disabled = False`. If that's true, then the redhat-operators source is enabled

Reply:
Added `disableAllDefaultSources` check